source/Applications/PointCloudTutorial.md

```diff
@@ -0,0 +1,131 @@
+|```Zivid::Array2D<Zivid::PointXYZW>```| ```PointCloud::copyPointsXYZW()``` or ```PointCloud::copyData<Zivid::PointXYZW>()```| 16 bytes |37 MB |
+|```Zivid::Array2D<Zivid::PointZ>```| ```PointCloud::copyPointsZ()``` or ```PointCloud::copyData<Zivid::PointZ>()```| 4 bytes |9 MB |
+|```Zivid::Array2D<Zivid::ColorRGBA>```| ```PointCloud::copyColorsRGBA()``` or ```PointCloud::copyData<Zivid::ColorRGBA>()```| 4 bytes |9 MB |
+|```Zivid::Image<Zivid::ColorRGBA>```| ```PointCloud::copyImageRGBA()```| 4 bytes |9 MB |
+|```Zivid::Array2D<Zivid::PointXYZColorRGBA>```| ```PointCloud::copyPointsXYZColorsRGBA()``` or ```PointCloud::copyData<PointXYZColorRGBA>()```| 16 bytes |37 MB |
+|```Zivid::Array2D<Zivid::PointXYZColorBGRA>```| ```PointCloud::copyPointsXYZColorsBGRA()``` or ```PointCloud::copyData<PointXYZColorBGRA>()```| 16 bytes |37 MB |
+|```Zivid::Array2D<Zivid::SNR>```| ```PointCloud::copySNRs()``` or ```PointCloud::copyData<Zivid::SNR>()```| 4 bytes |9 MB |
+
+#### Copy selected data from GPU to system memory (Zivid-allocated)
+
+If you are only concerned about e.g. RGB color data of the point cloud, you can copy only that data to the system memory.
+```cpp
+auto rgba = frame.pointCloud().copyColorsRGBA(); /* Colors are copied from the GPU and into a
+Zivid::Array2D<Zivid::ColorRGBA>, which takes ownership of the data. */
+
+auto *dataPtr = const_cast<void *>(static_cast<const void *>(rgba.data())); /* Cast the data pointer as
+a void*, since this is what the OpenCV matrix constructor requires. */
+
+cv::Mat rgba(rgba.height(), rgba.width(), CV_8UC4, dataPtr); /* We wrap this block of data in an OpenCV
+matrix. This is possible since the layout of Zivid::ColorRGBA exactly matches the layout of CV_8UC4.
+No copying occurs in this step. */
+```
+
+####  Copy selected data from GPU to system memory (user-allocated)
+
+In the above example, ownership of the data was held by the returned ```Zivid::Array2D<>``` objects. Alternatively, you may provide a pre-allocated memory buffer to ```Zivid::PointCloud::copyData(dataPtr)```; the type of ```dataPtr``` defines what shall be copied (```PointXYZ```, ```ColorRGBA```, etc.).
+
+Now let us look at the exact same use case as above. However, this time, we allow OpenCV to allocate the necessary storage. Then we ask the Zivid API to copy data directly from the GPU into this memory location.
+
+```cpp
+const auto pointCloud = frame.pointCloud(); /* Get a handle to the full point cloud on the GPU. */
+
+auto rgba = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); /* Allocate an OpenCV matrix with
+an appropriate size. */
+
+auto *dataPtr = reinterpret_cast<Zivid::ColorRGBA *>(rgba.data); /* Cast the OpenCV data pointer to
+ColorRGBA* so the Zivid API can understand which data to copy. */
+
+pointCloud.copyData(dataPtr); /* Copy RGBA data straight into the OpenCV memory buffer. */
```

Comment:
Maybe let the comment be "Copy RGBA data straight from GPU memory into the OpenCV memory buffer."
More accurate, and also sounds more impressive ;)

Reply:
done